File: app/src/main/java/com/idt/test_service/MyBroadcastReceiver.java
package com.idt.test_service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class MyBroadcastReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if(intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED)) {
            Log.e("MyBroadcastReceiver", "start serviceIntent");
            Intent serviceIntent = new Intent(context, SimpleService.class);
            Log.e("MyBroadcastReceiver", "mid serviceIntent");
            context.startForegroundService(serviceIntent);
            Log.e("MyBroadcastReceiver", "end serviceIntent");

        }
    }
}

File: app/src/main/java/com/idt/test_service/MyBroadcastReceiver.kt
package com.idt.test_service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class MyBroadcastReceiver: BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        val TAG = "MyBroadcastReceiver"
        val action = intent!!.action
        if (action == Intent.ACTION_BOOT_COMPLETED) {
            Log.e("MyBroadcastReceiver", "start serviceIntent")
            val serviceIntent = Intent(context, SimpleService::class.java)
            Log.e("MyBroadcastReceiver", "mid serviceIntent")
            context!!.startForegroundService(serviceIntent)
            Log.e("MyBroadcastReceiver", "end serviceIntent")
        }else{
            Log.e(TAG, "onReceive: $action" )
        }
    }
}